src/framework/theme/components/chips/_chips.component.theme.scss

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+@function chip-icon-bg($bg, $percentage: 14%) {
+  @return tint(nb-theme($bg), $percentage);
+}
+
+@function chip-icon-hover-bg($bg, $percentage: 14%) {
+  @return shade(nb-theme($bg), $percentage);
+}
+
+@mixin nb-chips-theme() {
+  nb-chip {
+    color: nb-theme(chip-fg);
+    background-color: nb-theme(chip-bg);
+    font-weight: nb-theme(font-weight-bolder);
+
+    &.removable-chip .remove-icon {
+      background-color: nb-theme(chip-icon-bg);
+    }
+
+    &.removable-chip .remove-icon:hover {
+      background-color: nb-theme(chip-icon-hover-bg);
+    }
+  }
+
+  nb-chip.active-chip {
+    background-color: nb-theme(chip-active-bg);
+    color: nb-theme(chip-active-fg);
+
+    &.removable-chip .remove-icon {
+      background-color: chip-icon-bg(chip-active-bg);
+    }
+
+    &.removable-chip .remove-icon:hover {
+      background-color: chip-icon-hover-bg(chip-active-bg);
+    }
+  }
+
+  nb-chip.disabled-chip {
+    background-color: nb-theme(chip-disabled-bg);
+    color: nb-theme(chip-disabled-fg);
+    border: 0.125rem solid nb-theme(chip-disabled-border);
+
+    &.removable-chip .remove-icon {
+      background-color: chip-icon-bg(chip-disabled-bg);
+    }
+
+    &.removable-chip .remove-icon:hover {
+      background-color: chip-icon-hover-bg(chip-disabled-bg);
+    }
+  }
+
+  nb-chip.primary-chip {
+    background-color: nb-theme(chip-primary-bg);
```

Comment:
we can move this block into a mixin, can't we? so that it only accepts color and then produces the css